### PR TITLE
Swimming burns stamina

### LINKED ISSUE #42056: Swimming not using stamina
# Describe the bug

Swimming skill 0 and yet i still managed to swim off a prison island while naked to shore fairly far away with no hit to stamina.

# Steps To Reproduce

1. Start on prison island

2. Remove all clothing 

3. Swim to shore from a prison island

# Expected behavior

That kind of swimming at swim skill 0 either shouldn't be possible, or should cause a major hit to stamina at the very least. 

Build 10820

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Features "Swimming burns stamina."

#### Purpose of change
Closes #42056.

#### Describe the solution
Added stamina burn when swimming. The formula is very simple and depends only on character level in athletics. The lower the level, the higher the stamina burn. The effect is pretty noticeable on athletics levels 0-2 and becomes almost unimportant on higher levels.
изображение

While working on fix for #42056, I also noticed that even with zero stamina your character is able to hold on the water infinitely without any problems. So I added a check for `winded` effect to determine if player would drown.

#### Describe alternatives you've considered
Add more or less stamina burn while swimming.

#### Testing
Made my character naked. Dived into a deep water. Checked for stamina burn. When my character became winded, check for sinking.

#### Additional context
None.

### CODE REVIEW

**Comment by Entity-y** [comment]
Is there a way to regain stamina while swimming, perhaps by floating on the back unmoving for a few turns?

**Comment by kevingranade** [comment]
Do you have any kind of source for the magnitude here? Intuitively a very large impact seems warranted as early levels are achieved, but reducing it to 2% at 5he highest levels seems excessive.

**Comment by Night-Pryanik** [comment]
>Is there a way to regain stamina while swimming, perhaps by floating on the back unmoving for a few turns?

Just wait by `.`.

**Comment by Night-Pryanik** [comment]
> Do you have any kind of source for the magnitude here?

No, it was just my feelings about what is "right".

> Intuitively a very large impact seems warranted as early levels are achieved, but reducing it to 2% at 5he highest levels seems excessive.

Oh, it's not the percentage here, it's flat amount of stamina burn. E.g. at zero level swimming is burning 100 points of stamina, at 1 level it's 67 points, etc.

**Comment by Kelenius** [comment]
How much is 100 points? How does that compare to walking and running?

**Comment by Night-Pryanik** [comment]
> How much is 100 points? How does that compare to walking and running?

100 is a lot compared to walking, as walking burns 15 points per move. I think that people with no swimming experience at all (0 level) will make many non-efficient and exhausting movements while trying to stay afloat, so this high value seems appropriate.

**Comment by Maleclypse** [comment]
> > How much is 100 points? How does that compare to walking and running?
> 
> 100 is a lot compared to walking, as walking burns 15 points per move. I think that people with no swimming experience at all (0 level) will make many non-efficient and exhausting movements while trying to stay afloat, so this high value seems appropriate.

Level 10 swimming imo would be "swim the english channel" level swimmers, so I would think anything within the range of "could maintain this stamina burn rate for 24 hours" would be appropriate.  English channel crossings take between 7 and 27 hours.

**Comment by Night-Pryanik** [comment]
> Level 10 swimming imo would be "swim the english channel" level swimmers, so I would think anything within the range of "could maintain this stamina burn rate for 24 hours" would be appropriate.

This is exactly what you're describing. On level 10 you could swim for hours and don't notice any significant stamina burn.

**Comment by Night-Pryanik** [comment]
While I'm here, I forbid sleeping while swimming.

File: src/character.cpp [hunk 1]
```diff
@@ -5789,6 +5789,12 @@ void Character::burn_move_stamina( int moves )
         }
     }
     burn_ratio += overburden_percentage;
+
+    ///\EFFECT_SWIMMING decreases stamina burn when swimming
+    if( get_map().has_flag( ter_furn_flag::TFLAG_DEEP_WATER, pos() ) ) {
+        burn_ratio += 100 / std::pow( 1.5, get_skill_level( skill_swimming ) );
```
Comment (kevingranade) [inline src/character.cpp:5795]:
```suggestion
        burn_ratio += 100 / std::pow( 1.1, get_skill_level( skill_swimming ) );
```

**Pull request merged**